### PR TITLE
fix: download car files from DO space in state migration unit tests

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

- Download car files from DO space in state migration unit tests
- Revive nv18 and nv19 unit tests
- Fix some DO space links to use CDN

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by lemmih — APPROVED** [review]

File: src/state_migration/tests/mod.rs [hunk 1]
```diff
@@ -29,19 +31,63 @@ async fn test_nv17_state_migration_calibnet() -> Result<()> {
...
 ) -> Result<()> {
-    // TODO: CompressedCarV1BackedBlockstore does not work for nv18 and nv19,
-    // use UncompressedCarV1BackedBlockstore instead
+    // Car files are cached under data folder for Go test to pick up without network access
+    let car_path = format!("./src/state_migration/tests/data/{old_state}.car");
+    if !Path::new(&car_path).is_file() {
+        let tmp: tempfile::TempPath = tempfile::NamedTempFile::new()?.into_temp_path();
+        {
+            let mut reader = crate::utils::net::reader(&format!(
+                " https://forest-snapshots.fra1.cdn.digitaloceanspaces.com/state/{old_state}.car"
```
Comment (LesnyRumcajs) [inline src/state_migration/tests/mod.rs:78]: Does it belong in the snapshots bucket? We can create new ones freely. How about a `forest-continuous-integration`?

Comment (hanabi1224) [reply src/state_migration/tests/mod.rs:78]: I can do that

Comment (LesnyRumcajs) [reply src/state_migration/tests/mod.rs:78]: Let's do it. I'd rather separate our "prod" buckets from dev ones.

Comment (hanabi1224) [reply src/state_migration/tests/mod.rs]: Fixed

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**